middleware/logging.go

```diff
@@ -14,11 +14,12 @@ import (
 
 // Log middleware logs http requests
 type Log struct {
-	Log                   logging.Interface
-	LogRequestHeaders     bool // LogRequestHeaders true -> dump http headers at debug log level
-	LogRequestAtInfoLevel bool // LogRequestAtInfoLevel true -> log requests at info log level
-	SourceIPs             *SourceIPExtractor
-	HttpHeadersToExclude  map[string]bool
+	Log                      logging.Interface
+	DisableRequestSuccessLog bool
```

Comment:
Good point. I can definitely rewrite it but I did it this way because:
- It is more consistent with the existing DisableRequestSuccessLog
- It allow `weaveworks/common` users to leave it unset by default (i.e: won't introduce an unwanted change)
That said, I'm happy with either; based on those two points, do you prefer `DisableRequestSuccessLog` (which is false by default) ot `LogSuccesfulRequests` (which is truthy by default)? If you prefer the former, would you like to see a rewrite of  `DisableRequestSuccessLog` too?

Reply:
> It is more consistent with the existing DisableRequestSuccessLog

Can you rephrase?  It looks to me like you are adding DisableRequestSuccessLog.

> leave it unset by default (i.e: won't introduce an unwanted change)

This doesn't apply to anyone calling `NewLogMiddleware`; do you think anyone is doing it another way?

Reply:
> > It is more consistent with the existing DisableRequestSuccessLog
> 
> Can you rephrase? It looks to me like you are adding DisableRequestSuccessLog.

I was referring to the existing `DisableRequestSuccessLog` defined under `server/server.go`.

> > leave it unset by default (i.e: won't introduce an unwanted change)
> 
> This doesn't apply to anyone calling `NewLogMiddleware`; do you think anyone is doing it another way?

Hmm I thought that mostly users were only doing `server.New()`, without defining `NewLogMiddleware` directly. I'll check how we do it, but tbh your suggestion of defining a flag on my program will make this work any way I prefer, so indeed I shouldn't be too concerned about it.

Reply:
OK, I didn't know there was already one in `middleware.GRPCServerLog`.
In that case, you can add one for `middleware.Log` and make it configurable in exactly the same way, no extra parameters.

Reply:
I'm not sure if that's what you had in mind but I pushed https://github.com/weaveworks/common/pull/284/commits/d9330659bfde5eaaa4e1877f3e56a39ed2dc2648 that does the following:
- Remove `DisableRequestSuccessLog` from the list of parameters, to avoid a breaking change to `NewLogMiddleware` users
- Make `LogMiddleware.DisableRequestSuccessLog` default value to be whatever is set on `server.Config.DisableRequestSuccessLog`.